### PR TITLE
[C++] Rename `rerun::ComponentBatch` to `rerun::Collection` (and related constructs)

### PR DESCRIPTION
**Pull request by Wumpf**
### What

Not much else has changed yet except some doc massaging. First step towards solving

* #3794

`rerun::Collection` will evolve from here on and be the corner piece of our zero copy & component adaptability strategy (just like ComponentBatch is already!)

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/4225) (if applicable)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4225)
- [Docs preview](https://rerun.io/preview/8cda4b927098a7f15ac746a54a257b448f6c28e3/docs) 
- [Examples preview](https://rerun.io/preview/8cda4b927098a7f15ac746a54a257b448f6c28e3/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: rerun_cpp/src/rerun/archetypes/bar_chart_ext.cpp [hunk 1]
```diff
@@ -1,12 +1,9 @@
 #include "../half.hpp"
 #include "bar_chart.hpp"
 
-// #define EDIT_EXTENSION
+namespace rerun::archetypes {
```
Comment (emilk) [inline rerun_cpp/src/rerun/archetypes/bar_chart_ext.cpp:4]: can this be added to the clang-format file?

File: rerun_cpp/src/rerun/archetypes/image_ext.cpp [hunk 2]
```diff
@@ -1,60 +1,70 @@
...
+            Error(
+                ErrorCode::InvalidTensorDimension,
+                "Only images with 1, 3 and 4 channels are supported."
+            )
+                .handle();
+            return;
+        }
+
+        // We want to change the dimension names if they are not specified.
+        // But rerun collections are strictly immutable, so create a new one if necessary.
```
Comment (emilk) [inline rerun_cpp/src/rerun/archetypes/image_ext.cpp:49]: Are we sure using `Collection` for `TensorDimension` makes sense? When will it avoid allocations and memcpys, realistically?

**Review by emilk — APPROVED** [review]

File: rerun_cpp/src/rerun/archetypes/image_ext.cpp (hunk 2 above)
Comment (Wumpf) [reply rerun_cpp/src/rerun/archetypes/image_ext.cpp:49]: it's less about avoiding copies & allocs but more about being able to use the adapters for this type as well

File: rerun_cpp/src/rerun/archetypes/bar_chart_ext.cpp (hunk 1 above)
Comment (Wumpf) [reply rerun_cpp/src/rerun/archetypes/bar_chart_ext.cpp:4]: Only found https://clang.llvm.org/docs/ClangFormatStyleOptions.html#compactnamespaces which isn't quite the same

But as you pointed out elsewhere there's also the option to not indent after namespace, we should try that (in a separate pr)
https://clang.llvm.org/docs/ClangFormatStyleOptions.html#namespaceindentation

**Comment by Wumpf** [comment]
oooops, that was the wrong source branch :flushed: :flushed:. The plan was to have a PR only for up to 3de46c8a5d294eb5ad68e5beb35a64d0610e992a 
The rest is still super experimental. Mopping up. Since this was already approved now I merge the rename pr and bring up a new one for the rest later!

**Pull request closed without merging**